### PR TITLE
ci: include Job ID and Go version in the cache key.

### PR DESCRIPTION
**Pull request by mathetake**
Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

### CODE REVIEW

File: .github/workflows/commit.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
           path: |
             ~/.cache/go-build
             ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}-20211231
+          key: ${{ runner.os }}-go-${{ env.GO_VERSION }}-${{ hashFiles('**/go.sum') }}
```
Comment (mathetake) [inline .github/workflows/commit.yaml:33]: we should use different key than benches as that is the main cause of golangci-lint cache non-existence

**Comment by mathetake** [comment]
strange `{{ env.GITHUB_JOB }}` seems empty..

**Comment by codefromthecrypt** [comment]
maybe you can't reference by env.X in this scope?

**Comment by mathetake** [comment]
yeah got it https://david.gardiner.net.au/2021/07/github-actions-default-environment-variables.html

**Review by codefromthecrypt — APPROVED** [review]

**Pull request merged**